### ISSUE TITLE
Non-Latin Characters on Windows

### ISSUE THREAD

**Issue by robertguetzkow** [robertguetzkow/libtrashcan#2]
It seems that the the char* to wchar_t* conversion doesn't work properly for non latin characters.

**Comment by robertguetzkow** [comment]
~After investigating this doesn't seem to be a problem of the conversion from multibyte to wide characters, but rather with `SHCreateItemFromParsingName()`~. When using `MultiByteToWideChar()` instead of `mbstowcs_s()` or even using `wmain` and passing in the `wchar_t *` directly, this function doesn't seem to parse non-latin characters such as Russian or Chinese.

Edit: `SHCreateItemFromParsingName()` isn't at fault here. This is a problem caused by the encoding of the argument and the conversion to wide characters.

**Comment by robertguetzkow** [comment]
What does work is using `wmain` and setting the codepage of powershell to UTF-8. So this problem seems to be a combination of encoding problems starting from the encoding supplied by the command line.

There is probably something unexpected happening when using, e.g. an UTF-8 encoded argument and using `char **argv`.

**Comment by robertguetzkow** [comment]
So the problem appears to be that Powershell and cmd aren't able to pass the Unicode or UTF-8 encoded string properly to the program as argument when `int main(int argc, char **argv)` is used. Instead of using the char array as a sequence of bytes and storing the encoded string in it, every character outside the range of a single char is converted to a `?` or `0xF3`. Therefore one cannot recover the wide characters. This is the behavior after setting `chcp 65001` and input and output encoding to UTF-8.

```
"./example.exe" "D:\123\test\test\abcüäö%&Дд.txt"
using int main(int argc, char **argv): 
443A5C3132335C746573745C746573745C616263FCE4F625263F3F2E747874
using int main(int argc, char **argv) converted to wchar_t*: 
0044003A005C003100320033005C0074006500730074005C0074006500730074005C00610062006300FC00E400F600250026003F003F002E007400780074
using wmain(int argc, wchar_t **argv): 
0044003A005C003100320033005C0074006500730074005C0074006500730074005C00610062006300FC00E400F60025002604140434002E007400780074
```

**Comment by robertguetzkow** [comment]
In other words there are two things that don't work properly. Under some configurations Windows omits characters from the string, thus resulting in a path that doesn't exist and `mbstowcs_s` doesn't work properly for converting `char*` to `wchar_t*` because it has no information about the current encoding.

**Comment by robertguetzkow** [comment]
This issue is resolved by 37bc6ceb136bd333f4128dc7204f26ce31eab6fe. libtrashcan now requires UTF-8 encoding for `int soft_delete(const char *path);` on all platforms. If a specific code page needs to be used on Windows, you can configure it using `int soft_delete_com(const char *path, unsigned int code_page, bool init_com);`. 

Windows 10 offers a beta feature to activate UTF-8:

1. Open the Windows Control Panel
2. Open the "Time and Region" settings
3. Open the Region settings
4. Switch to the "Administrative" tab
5. Press the "Change system locale"
6. Enable UTF-8
